containers/daap-query-data/src/var/task/query_data.py

```diff
@@ -0,0 +1,124 @@
+                QueryExecutionContext={"Database": database_name},
+                WorkGroup="data_product_workgroup",
+            )
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "InvalidRequestException":
+                raise ValueError(e)
+            else:
+                raise ValueError(e)
+
+        query_id = res["QueryExecutionId"]
+        while response := athena_client.get_query_execution(
+            QueryExecutionId=query_id
+        ):
+            state = response["QueryExecution"]["Status"]["State"]
+            if state not in ["SUCCEEDED", "FAILED"]:
+                time.sleep(0.1)
+            else:
+                break
+
+        if not state == "SUCCEEDED":
+            logger.error(
+                "Query {}, failed with response: {}".format(
+                    sql,
+                    response["QueryExecution"]["Status"].get("StateChangeReason"),
+                )
+            )
+            raise ValueError(
+                response["QueryExecution"]["Status"].get("StateChangeReason")
+            )
+
+        return query_id
+
+
+def format_athena_results_to_table(results):
+    # Format the results into a table string
+    if not results:
+        return "No data to display"
+
+    table = ''
+
+    # Data rows
+    headers= [value for header_dict in results['ResultSet']['Rows'][0]['Data'] for value in header_dict.values()]
+    print(headers)
+    column_widths = {header: len(header) for header in headers}
+
+    # Data rows
+    data_result=[]
+    for row in results['ResultSet']['Rows'][1:]:
+        data = [value for data_dict in row['Data'] for value in data_dict.values()]
+        data_result.append({key: value for key, value in zip(headers, data)})
+
+    # Calculate the maximum width for each column based on header and data values
+    for row in data_result:
+        for header in headers:
+            value = row[header]
+            column_widths[header] = max(column_widths[header], len(str(value)))
+
+    # Build the table header
+    table = "|"
+    for header in headers:
+        table += f" {header.ljust(column_widths[header])} |"
+    table += "\n"
+
+    # Add data rows
+    for row in data_result:
+        table += "|"
+        for header in headers:
+            value = row[header]
+            table += f" {str(value).ljust(column_widths[header])} |"
+        table += "\n"
+
+    return table
+
+def handler(event,context,athena_client=athena_client):
+    data_product_name = event["pathParameters"]["data-product-name"]
+    table_name = event["pathParameters"]["table-name"]
+
+    logger = DataPlatformLogger(
+        data_product_name=data_product_name,
+        extra={
+            "image_version": os.getenv("VERSION", "unknown"),
+            "base_image_version": os.getenv("BASE_VERSION", "unknown"),
+            "table_name": table_name,
+        },
+    )
+
+    query = f'SELECT * FROM "{data_product_name}"."{table_name}" LIMIT 10'
```

Comment:
Should we also order by execution timestamp descending? Doesn't really matter for now, I'm just thinking ahead to partial loads

Reply:
I think the query should be for latest partition, as the use case was for someone to check the data they just loaded had flowed through, so add
`where load_timestamp = (select max(load_timestamp) from {data_product_name}.{table_name})`

Reply:
No it doesn't, need to specify in query

Reply:
In Amazon Athena, when you use a query like SELECT * FROM table_name, Athena will return data from the latest partition, assuming the table you're querying is a partitioned table and it's set up to use partitions based on a date or other criteria. If you have a table with partitions based on time (e.g., date), and you don't specify a specific partition in your query, Athena will query the partition with the latest data.